File: 백준/Silver/1920. 수 찾기/수 찾기.py
import sys
input = sys.stdin.readline

n = int(input())
arr_n = list(map(int, input().split()))
arr_n.sort()

m = int(input())
arr_m = tuple(map(int, input().split()))

def binary_search(num, start, end):
    while start <= end:
        mid = (start + end) // 2

        if arr_n[mid] == num:
            return 1

        elif arr_n[mid] > num:
            end = mid - 1

        else: # arr_n[mid] < num
            start = mid + 1

    return 0

for num in arr_m:
    print(binary_search(num, 0, n-1))

File: 백준/Silver/1920. 수 찾기/수 찾기.java
import java.io.*;
import java.util.*;

public class Main {

    private static int[] A;
    private static int N;

    public static void main(String[] args) throws Exception{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        N = Integer.parseInt(br.readLine());
        A = new int[N];
        StringTokenizer st = new StringTokenizer(br.readLine());
        for(int i=0; i<N; i++){
            A[i] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(A);

        int M = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());
        StringBuilder sb = new StringBuilder();
        while(st.hasMoreTokens()){
            sb.append(binarySearch(Integer.parseInt(st.nextToken()))).append("\n");
        }
        System.out.println(sb.toString());
    }

    public static int binarySearch(int target){
        int st = 0;
        int en = N-1;
        while(st<=en){
            int mid = (st+en)/2;
            if(target>A[mid]){
                st = mid+1;
            }else if(target<A[mid]){
                en = mid-1;
            }else{
                return 1;
            }
        }
        return 0;
    }
}
